datalad/core/distributed/clone.py

```diff
@@ -564,6 +564,15 @@ def clone_dataset(
                 message=str(e),
                 **result_props,
             )
+
+            # We were supposed to clone a particular version but failed to.
+            # This is particularly pointless in case of subdatasets and
+            # potentially fatal with current implementation of recursion.
+            # see gh-5387
+            lgr.debug("removing failed clone attempt at %s", dest_path)
```

Comment:
Please make debug message more specific and actually more factually correct somewhat since it succeeded to clone, but failed to find corresponding commit. I even wonder if worth a warning level, but if keeping it debug. e.g.
```suggestion
            lgr.debug("Failed to checkout %s, removing this clone attempt at %s", checkout_gitsha, dest_path)
```

Reply:
Warning level shouldn't happen at that level, I think. `clone`/`get`/`install` may just proceed with the next candidate URL.